src/compiler/transformers/es2015.ts

```diff
@@ -1571,6 +1578,78 @@ namespace ts {
                         break;
                 }
             }
+
+            /**
+             * Lazily creates the way class members will access the class prototype.
+             */
+            function getPrototypeStorageName() {
+                if (prototypeStorageName) {
+                    return prototypeStorageName;
+                }
+
+                const originalPrototypeAccess = createPropertyAccess(getInternalName(node), "prototype");
+
+                // If the class has exactly one non-static member, it'll access prototype members on itself:
+                //   ClassName.prototype.member = ...
+                if (!membersContainMultipleUniqueNames(node.members.filter(classMemberAssignsToPrototype))) {
+                    prototypeStorageName = originalPrototypeAccess;
+                    return prototypeStorageName;
+                }
+
+                // Since it has multiple non-static members, it'll store that prototype as a variable that can be minified:
+                //  var proto = ClassName.prototype;
+                //  proto.memberOne = ...
+                //  proto.memberTwo = ...
+                prototypeStorageName = createUniqueName("proto");

```

Comment:
This has the downside of reducing readability if you are debugging the JavaScript of a large class without source maps. I'd recommend you consider naming the variable something like `ClassName_prototype` (if the class has a usable name). Since the goal of this change is to assist minifiers, a long name like `ClassName_prototype` would be minified anyways.

I'd also recommend you use `createUniqueName(candidateName, GeneratedIdentifierFlags.Optimistic)` to avoid the `_1` suffix for cases where there's no collision.